plugins/wallet_plugin/wallet.cpp

```diff
@@ -211,205 +309,268 @@ class soft_wallet_impl
 }
 
-void soft_wallet::encrypt_keys()
-{
+void soft_wallet::encrypt_keys() {
    my->encrypt_keys();
 }
 
-void soft_wallet::lock()
-{ try {
-   SYS_ASSERT( !is_locked(), wallet_locked_exception, "Unable to lock a locked wallet" );
-   encrypt_keys();
-   for( auto key : my->_keys )
-      key.second = private_key_type();
-
-   my->_keys.clear();
-   my->_checksum = fc::sha512();
-} FC_CAPTURE_AND_RETHROW() }
-
-void soft_wallet::unlock(string password)
-{ try {
-   FC_ASSERT(password.size() > 0);
-   auto pw = fc::sha512::hash(password.c_str(), password.size());
-   vector<char> decrypted = fc::aes_decrypt(pw, my->_wallet.cipher_keys);
-   auto pk = fc::raw::unpack<plain_keys>(decrypted);
-   FC_ASSERT(pk.checksum == pw);
-   my->_keys = std::move(pk.keys);
-   my->_checksum = pk.checksum;
-} SYS_RETHROW_EXCEPTIONS(chain::wallet_invalid_password_exception,
-                          "Invalid password for wallet: \"${wallet_name}\"", ("wallet_name", get_wallet_filename())) }
-
-void soft_wallet::check_password(string password)
-{ try {
-   FC_ASSERT(password.size() > 0);
-   auto pw = fc::sha512::hash(password.c_str(), password.size());
-   vector<char> decrypted = fc::aes_decrypt(pw, my->_wallet.cipher_keys);
-   auto pk = fc::raw::unpack<plain_keys>(decrypted);
-   FC_ASSERT(pk.checksum == pw);
-} SYS_RETHROW_EXCEPTIONS(chain::wallet_invalid_password_exception,
-                          "Invalid password for wallet: \"${wallet_name}\"", ("wallet_name", get_wallet_filename())) }
-
-void soft_wallet::set_password( string password )
-{
-   if( !is_new() )
-      SYS_ASSERT( !is_locked(), wallet_locked_exception, "The wallet must be unlocked before the password can be set" );
-   my->_checksum = fc::sha512::hash( password.c_str(), password.size() );
+void soft_wallet::set_key_name(string current_key_name, string new_key_name) {
+   my->set_key_name(current_key_name, new_key_name);
+}
+
+void soft_wallet::set_key_name(private_key_type private_key, string key_name) {
+   my->set_key_name(private_key, key_name);
+}
+
+void soft_wallet::set_key_name(public_key_type public_key, string key_name) {
+   my->set_key_name(public_key, key_name);
+}
+
+string soft_wallet::generate_key_name(string prefix, string suffix) {
+   return my->generate_key_name(prefix, suffix);
+}
+
+void soft_wallet::lock() {
+   try {
+      SYS_ASSERT(!is_locked(), wallet_locked_exception, "Unable to lock a locked wallet");
+      encrypt_keys();
+      for (auto key : my->_keys)
+         key.second = private_key_type();
+
+      my->_keys.clear();
+      my->_key_by_name.clear();
+      my->_checksum = fc::sha512();
+   } FC_CAPTURE_AND_RETHROW()
+}
+
+void soft_wallet::unlock(string password) {
+   try {
+      FC_ASSERT(password.size() > 0);
+      auto         pw        = fc::sha512::hash(password.c_str(), password.size());
+      vector<char> decrypted = fc::aes_decrypt(pw, my->_wallet.cipher_keys);
+      switch (my->_wallet.version) {
```

Comment:
I don't think we need to maintain two versions. Just go with the new one.

Reply:
Corect, we're starting from nil, I agree.  kk rebasing now